### PR TITLE
fix: source-map matching in React Native DevTools

### PR DESCRIPTION
**Pull request by jbroma**
### Summary

Fix bad source-map matching in DevTools by aligning the following:

- [x] - add namespace to `SourceMapDevToolPlugin`
- [x] - align `sourceURL` that gets passed to JSI `evaluateJavascript`

Before:
image

After:
image

### Test plan

- [x] - proper mapping in DevTools in testers on iOS
- [x] - proper mapping in DevTools in testers on Android

### CODE REVIEW

File: packages/repack/android/src/main/java/com/callstack/repack/ScriptConfig.kt [hunk 1]
```diff
@@ -33,13 +35,25 @@ data class ScriptConfig(
...
 
-            val url = URL(
-                    if (query != null) {
-                        "$urlString?$query"
-                    } else {
-                        urlString
-                    }
-            )
+            val initialUrl = URL(urlString)
+            val initialUri = initialUrl.toURI()
```
Comment (okwasniewski) [inline packages/repack/android/src/main/java/com/callstack/repack/ScriptConfig.kt:39]: nit: its almost impossible to distinguish those two variable names 🤣

File: packages/repack/ios/ScriptConfig.h [hunk 2]
```diff
@@ -20,6 +20,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (readonly) NSNumber *timeout;
 @property (readonly) NSString *verifyScriptSignature;
 @property (readonly) NSString *uniqueId;
+@property (readonly) NSString *sourceUrl;
```
Comment (okwasniewski) [inline packages/repack/ios/ScriptConfig.h:23]: nit: a further improvement, you can probably make all of those properties nonatomic, https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjectiveC/Chapters/ocProperties.html

**Review by okwasniewski — APPROVED** [review]

File: packages/repack/android/src/main/java/com/callstack/repack/ScriptConfig.kt (hunk 1 above)
Comment (jbroma) [reply packages/repack/android/src/main/java/com/callstack/repack/ScriptConfig.kt:39]: damn, good point 👍

File: packages/repack/ios/ScriptConfig.h (hunk 2 above)
Comment (jbroma) [reply packages/repack/ios/ScriptConfig.h:23]: what would be the benefit of that?

**Comment by jbroma** [comment]
> Could you document the addition of sourceUrl somewhere?

It's not an addition per se - we should do a whole doc page on using DevTools and dealing with issues - the only thing that user needs to do here is to remember to set `output.uniqueName`

File: packages/repack/ios/ScriptConfig.h (hunk 2 above)
Comment (okwasniewski) [reply packages/repack/ios/ScriptConfig.h:23]: _speed_ 🚅, if you add nonatomic the access should be faster

Comment (jbroma) [reply packages/repack/ios/ScriptConfig.h:23]: will do in a separate PR to keep this on topic, thx!

**Pull request merged**